src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml

```diff
@@ -400,11 +400,12 @@ let setup_daemon logger =
   and node_error_url =
     flag "--node-error-url" ~aliases:[ "node-error-url" ] (optional string)
       ~doc:"URL of the node error collection service"
-  and simplified_node_stats =
-    flag "--simplified-node-stats"
-      ~aliases:[ "simplified-node-stats" ]
-      (optional_with_default true bool)
-      ~doc:"whether to report simplified node stats (default: true)"
+  and node_status_type =
+    flag "--node-status-type" ~aliases:[ "node-status-type" ]
+      (optional_with_default `Simple Cli_lib.Arg_type.node_status_type)
+      ~doc:
+        "full|simple|none Whether to report simple, full or no node stats \
+         (default: simple)"
```

Comment:
The default should be `none`. We don't want opt-out surveillance, users really don't like it.

Reply:
@mrmr1993 Correct - this is the general rule along which we build our systems. However, in this case we were advised to implement default reporting only for the soft-fork mainnet version. It will not go to Berkeley. It is only here to improve HF procedure. We do not report data like IP address in the default simple mode.

Reply:
> We do not report data like IP address in the default simple mode.

You use an http connection to report the status data, so you're going to be explicitly reporting the IP address to the server that way.

I still don't think this is a good look: "hey, here's the release to prepare for the upgrade, and by the way we've also enabled surveillance in it".

> However, in this case we were advised to implement default reporting only for the soft-fork mainnet version.

If we do align to do this, this should be accompanied by a PR against `berkeley` that merges this PR there and then reverts these changes, to ensure that it doesn't get introduced by a merge-up.

Reply:
Maybe a better API is a new flag `--enable-hardfork-reporting` that overrides the node status type to `\`Simple` and uses the default URL. The purpose of the flag is self documenting.

Reply:
Thanks for raising the privacy concerns @mrmr1993!

For wider context, the stats are important to make sure we know enough stake is online to proceed with the HF. Across a quorum of MF and o1Labs we felt it was acceptable trade-off to report stats by default as long as:

- It's only for the pre-HF release, it won't be default going forward
- It's easily disabled `--node-status-type=none`
- It's communicated with the release

Reply:
To clarify pre-HF means pre-Hardfork and the change is only included in this temporary softfork.